From b030496c64d8b93cb1bf962c8e9a8d13937fef7a Mon Sep 17 00:00:00 2001
From: Joseph Kogut <joseph.kogut@gmail.com>
Date: Wed, 2 Mar 2022 15:11:22 -0800
Subject: [PATCH 1/1] Revert "overlays: i2c-gpio: Avoid open-drain warnings"

On the Balena Fin (customized with depopulated pca9633 LED controller) with a CM3+, setting the GPIO_OPEN_DRAIN flag on the GPIOs doesn't pull the SDA/SCL lines high as expected during inactivity, breaking software i2c. It works as expected without this flag.

Relevant issue: https://github.com/balena-os/balena-raspberrypi/issues/640

This reverts commit 129d946f03781376187acefad1cd15f5be19766e.

Signed-off-by: Joseph Kogut <joseph.kogut@gmail.com>
---
 arch/arm/boot/dts/overlays/balena-fin-overlay.dts   | 5 +----
 arch/arm/boot/dts/overlays/i2c-gpio-overlay.dts     | 6 ++----
 arch/arm/boot/dts/overlays/i2c-rtc-gpio-overlay.dts | 4 ++--
 3 files changed, 5 insertions(+), 10 deletions(-)

diff --git a/arch/arm/boot/dts/overlays/balena-fin-overlay.dts b/arch/arm/boot/dts/overlays/balena-fin-overlay.dts
index e7ead7cdf5f5..249c8202b2ed 100644
--- a/arch/arm/boot/dts/overlays/balena-fin-overlay.dts
+++ b/arch/arm/boot/dts/overlays/balena-fin-overlay.dts
@@ -1,8 +1,6 @@
 /dts-v1/;
 /plugin/;
 
-#include <dt-bindings/gpio/gpio.h>
-
 /{
 	compatible = "brcm,bcm2835";
 
@@ -50,8 +48,7 @@ power_ctrl: power_ctrl {
 
 			i2c_soft: i2c@0 {
 				compatible = "i2c-gpio";
-				gpios = <&gpio 43 (GPIO_ACTIVE_HIGH|GPIO_OPEN_DRAIN) /* sda */
-				         &gpio 42 (GPIO_ACTIVE_HIGH|GPIO_OPEN_DRAIN) /* scl */>;
+				gpios = <&gpio 43 0 /* sda */ &gpio 42 0 /* scl */>;
 				i2c-gpio,delay-us = <5>;
 				i2c-gpio,scl-open-drain;
 				i2c-gpio,sda-open-drain;
diff --git a/arch/arm/boot/dts/overlays/i2c-gpio-overlay.dts b/arch/arm/boot/dts/overlays/i2c-gpio-overlay.dts
index 63231b5d7c0c..39e7bc5fa9d8 100644
--- a/arch/arm/boot/dts/overlays/i2c-gpio-overlay.dts
+++ b/arch/arm/boot/dts/overlays/i2c-gpio-overlay.dts
@@ -2,8 +2,6 @@
 /dts-v1/;
 /plugin/;
 
-#include <dt-bindings/gpio/gpio.h>
-
 / {
 	compatible = "brcm,bcm2835";
 
@@ -14,8 +12,8 @@ __overlay__ {
 			i2c_gpio: i2c@0 {
 				reg = <0xffffffff>;
 				compatible = "i2c-gpio";
-				gpios = <&gpio 23 (GPIO_ACTIVE_HIGH|GPIO_OPEN_DRAIN) /* sda */
-					 &gpio 24 (GPIO_ACTIVE_HIGH|GPIO_OPEN_DRAIN) /* scl */
+				gpios = <&gpio 23 0 /* sda */
+					 &gpio 24 0 /* scl */
 					>;
 				i2c-gpio,delay-us = <2>;        /* ~100 kHz */
 				#address-cells = <1>;
diff --git a/arch/arm/boot/dts/overlays/i2c-rtc-gpio-overlay.dts b/arch/arm/boot/dts/overlays/i2c-rtc-gpio-overlay.dts
index c83480c1c327..4252b0c4d987 100644
--- a/arch/arm/boot/dts/overlays/i2c-rtc-gpio-overlay.dts
+++ b/arch/arm/boot/dts/overlays/i2c-rtc-gpio-overlay.dts
@@ -13,8 +13,8 @@ fragment@100 {
 		__overlay__ {
 			i2cbus: i2c-gpio-rtc@0 {
 				compatible = "i2c-gpio";
-				gpios = <&gpio 23 (GPIO_ACTIVE_HIGH|GPIO_OPEN_DRAIN) /* sda */
-					 &gpio 24 (GPIO_ACTIVE_HIGH|GPIO_OPEN_DRAIN) /* scl */
+				gpios = <&gpio 23 0 /* sda */
+					 &gpio 24 0 /* scl */
 					>;
 				i2c-gpio,delay-us = <2>;        /* ~100 kHz */
 				#address-cells = <1>;
-- 
2.35.1

